src/Symfony/Component/Mime/SMimeMessage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime;
+
+/**
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * @experimental in 4.3
+ */
+class SMimeMessage extends RawMessage
+{
+    private $originalMessage;
+
+    /**
+     * @param iterable|string $message
+     */
+    public function __construct(RawMessage $originalMessage, $message)
```

Comment:
I would swap the two arguments here as the most important one is the signed/encrypted message.